src/stream/src/common/cache/top_n_cache.rs

```diff
@@ -51,21 +62,41 @@ impl<K: Ord, V> TopNCache<K, V> {
     /// Clear the cache.
     pub fn clear(&mut self) {
         self.entries.clear();
+        self.kv_heap_size = 0;
     }
 
     /// Insert an entry into the cache.
     pub fn insert(&mut self, key: K, value: V) -> Option<V> {
+        let key_size = key.estimated_heap_size();
```

Comment:
It's not accurate here. The `insertion` and `deletion` may introduce btree's split and merge, which may cause more memory allocation and deallocation.

Reply:
https://github.com/risingwavelabs/risingwave/pull/9518/files/1a862b36dada23c4e729655d1f1992438773b425#diff-2ca07a1d0befc8f614636892de04c494d71d8a4dd3dc43667943f1d100991d4eR30-R31